src/core/expression/qgsexpressioncontextutils.cpp

```diff
@@ -975,3 +975,99 @@ QgsScopedExpressionFunction *QgsExpressionContextUtils::GetLayerVisibility::clon
   func->mScaleBasedVisibilityDetails = mScaleBasedVisibilityDetails;
   return func;
 }
+
+//
+// mesh expression context
+//
+
+/// @cond PRIVATE
+class CurrentVertexZValueExpressionFunction: public QgsScopedExpressionFunction
+{
+  public:
+    CurrentVertexZValueExpressionFunction():
+      QgsScopedExpressionFunction( "$vertex_z",
+                                   0,
+                                   QObject::tr( "Meshes" ),
+                                   QObject::tr( "return the z value of the current vertex" ), false )
```

Comment:
I think the help text can be left out as it is provided in the dedicated expression help file.